tests/unit/test_eurostat.py

```diff
@@ -0,0 +1,223 @@
+"""'test_eurostat.py'."""
+
+import json
+import logging
+
+import pandas as pd
+import pytest
+from viadot.sources import Eurostat
+
+
+class EurostatMock(Eurostat):
+    """Mock of Eurostat source class."""
+
+    def __init__(self, dataset_code=None):
+        # Inicjalizacja bez dataset_code
+        super().__init__(dataset_code=dataset_code)
+
+    def _download_json(self):
+        with open("test_eurostat_response.json") as file:
+            data = json.load(file)
+            return data
+
+
+def test_eurostat_dictionary_to_df():
+    """Test eurostat_dictionary_to_df method from source class."""
+    eurostat = EurostatMock(dataset_code="")  # Możesz przekazać pusty string lub None
+    data = eurostat._download_json()
+
+    result_df = eurostat.eurostat_dictionary_to_df(["geo", "time"], data)
+
+    assert list(result_df.columns) == ["geo", "time", "indicator"]
+
+    expected_years = ["2020", "2021", "2022"]
+    assert result_df["time"].unique().tolist() == expected_years
+
+    expected_geo = ["Germany", "France", "Italy"]
+    assert result_df["geo"].unique().tolist() == expected_geo
+
+    expected_indicator = [
+        100,
+        150,
+        200,
+        110,
+        160,
+        210,
+        120,
+        170,
+        220,
+    ]
+    assert result_df["indicator"].unique().tolist() == expected_indicator
+
+URL = (
+    "https://ec.europa.eu/eurostat/api/dissemination/statistics/1.0"
+    "/data/ILC_DI04?format=JSON&lang=EN"
+)
+
+
+def test_wrong_dataset_code_logger(caplog):
+    """Tests that the error logging feature correctly.
+
+    Identifies and logs errors for incorrect dataset codes.
+    """
+    task = Eurostat(dataset_code="ILC_DI04E")
+
+    with pytest.raises(ValueError, match="DataFrame is empty!"):
+        with caplog.at_level(logging.ERROR):
+            task.to_df()
+    assert (
+        "Failed to fetch data for ILC_DI04E, please check correctness of dataset code!"
+        in caplog.text
+    )
+
+
+def test_and_validate_dataset_code_without_params(caplog):
+    """Tests that the data retrieval feature returns a non-empty DataFrame.
+
+    For a valid dataset code
+    """
+    task = Eurostat(dataset_code="ILC_DI04").to_df()
```

Comment:
It looks like this test doesn't use the mocked source, so it's not a unit test. You need to rewrite it (and other tests below which also do the same) so that it doesn't actually connect to the API or move all these integration tests into `tests/integration` directory.

Reply:
[7824be9](https://github.com/dyvenia/viadot/pull/1041/commits/7824be98f76a5daf19f8dadf9eba3c8147bbdee8)